tools/api_protolock/protolock_test.py

```diff
@@ -0,0 +1,149 @@
+
+DEFAULT_PROTOLOCK_ARGS = ["--plugins="]
+
+
+class ProtolockTests(unittest.TestCase):
+
+    def setUp(self):
+        # create a temporary directory to house the proto.lock file and current proto file
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.protolock_path = runfiles.Create().Rlocation(
+            "com_github_nilslice_protolock/protolock_/protolock")
+
+    def tearDown(self):
+        self.temp_dir.cleanup()
+
+    def run_protolock_test(self, testtype, testname, additional_args=None, expect_no_changes=False):
+        # there's probably a better/safer way to navigate
+        # but it seems like check_spelling_pedantic_test.py takes this approach as well
+        tests_path = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), "..", "testdata", "api_protolock",
+            testtype)
+        protolock_args = DEFAULT_PROTOLOCK_ARGS + [
+            f"--protoroot={self.temp_dir.name}", f"--lockdir={self.temp_dir.name}"
+        ]
+        if additional_args is not None:
+            protolock_args.extend(additional_args)
+
+        # check that test files exist
+        current = os.path.join(tests_path, f"{testname}_current")
+        changed = os.path.join(tests_path, f"{testname}_next")
+        self.assertTrue(os.path.isfile(current))
+        self.assertTrue(os.path.isfile(changed))
+
+        # 1) copy start file into temp dir
+        # 2) protolock init
+        # 3) copy changed file into temp dir
+        # 4) protolock commit
+        # 5) check for differences (if changes are breaking, there should be none)
+
+        target = os.path.join(self.temp_dir.name, f"{testname}.proto")
+        copyfile(current, target)
+
+        # TODO: change this to use tools/run_command.py?
+        initial_result = subprocess.run([self.protolock_path, "init", *protolock_args],
+                                        capture_output=True)
+
+        self.assertEqual(len(initial_result.stdout), 0)
+        self.assertEqual(len(initial_result.stderr), 0)
+
+        lock_location = os.path.join(self.temp_dir.name, "proto.lock")
+        with open(lock_location) as f:
+            initial_lock = f.readlines()
+
+        copyfile(changed, target)
+
+        final_result = subprocess.run([self.protolock_path, "commit", *protolock_args],
+                                      capture_output=True)
+        with open(lock_location) as f:
+            post_lock = f.readlines()
+
+        if testtype == "breaking":
+            self.assertGreater(len(final_result.stdout), 0)
+            self.assertRegex(str(final_result.stdout), r"CONFLICT")
+
+            self.assertListEqual(initial_lock, post_lock)
+        elif testtype == "allowed":
+            self.assertEqual(len(final_result.stdout), 0)
+            self.assertEqual(len(final_result.stderr), 0)
+
+            if not expect_no_changes:
+                self.assertGreater(len(post_lock), 0)
+                self.assertNotEqual(initial_lock, post_lock)
```

Comment:
Should there be an else here to verify that `inital_lock` and `post_lock` are equal?